File: mvp-framework/src/main/java/com/stephen/mvpframework/utils/MemoryShareDataUtil.java
package com.stephen.mvpframework.utils;

import java.util.HashMap;
import java.util.Map;

/**
 * 临时储存
 * Created by Stephen on 2018/1/26.
 * StephenYoung0406@hotmail.com
 * <(￣ c￣)y▂ξ
 */

public class MemoryShareDataUtil {
    private final static Map<String, Object> dataMap = new HashMap<>();

    //清空内存
    public static void clear() {
        dataMap.clear();
    }

    public static boolean isHasData(String code) {
        return dataMap.size() > 0 && dataMap.containsKey(code);
    }

    public static Object getObj(String code) {
        return dataMap.get(code);
    }

    public static void setObj(Object obj) {
        dataMap.put("", obj);
    }

    public static void setObj(String code, Object obj) {
        dataMap.put(code, obj);
    }
}

File: mvp-framework/src/main/java/com/stephen/mvpframework/utils/MemoryShareDataUtil.kt
package com.stephen.mvpframework.utils

import java.util.HashMap

/**
 * 临时储存工具类
 * Created by Stephen on 2018/11/16.
 * StephenYoung0406@hotmail.com
 * <(￣ c￣)y▂ξ
 */
object MemoryShareDataUtil {
    private val dataMap = HashMap<String, Any>()

    //清空内存
    fun clear() {
        dataMap.clear()
    }

    fun isHasData(code: String): Boolean {
        return dataMap.size > 0 && dataMap.containsKey(code)
    }

    fun getObj(code: String): Any? {
        return dataMap[code]
    }

    fun setObj(obj: Any) {
        dataMap[""] = obj
    }

    fun setObj(code: String, obj: Any) {
        dataMap[code] = obj
    }
}